numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -4169,6 +4169,37 @@ initialize_casting_tables(void)
     }
 }
 
+#ifndef NPY_PY3K
+/*
+ * In python2, the `float` and `complex` types still implement the obsolete
+ * "tp_print" method, which uses CPython's float-printing routines to print the
+ * float.  Numpy's float_/cfloat inherit from Python float/complex, but
+ * override its tp_repr and tp_str methods. In order to avoid an inconsistency
+ * with the inherited tp_print, we need to override it too.
+ *
+ * In python3 the tp_print method is reserved/unused.
+ */
+static int
+float_print(PyObject *o, FILE *fp, int flags)
```

Comment:
Please pick a name that differs from the name used by the builtin float type.  Something like `doubletype_print` would be a reasonable choice and consistent with the other PyDoubleArrType_Type methods.